tpch - accelerated/s3[parquet]-turso[file].yaml benchmark failures

## Describe the bug

https://github.com/spiceai/spiceai/actions/runs/20677590284/job/59367324380

```
2026-01-03T13:09:35.841358Z  INFO spiced: Starting runtime v1.11.0-unstable+models
2026-01-03T13:09:35.845241Z  INFO runtime::init::caching: Initialized sql results cache; max size: 128.00 MiB, item ttl: 1s, hashing algorithm: XXH3, encoding: none
2026-01-03T13:09:35.845325Z  INFO runtime::init::caching: Initialized search results cache; max size: 128.00 MiB, item ttl: 1s
2026-01-03T13:09:35.845339Z  INFO runtime::init::caching: Initialized embeddings cache; max size: 128.00 MiB, item ttl: 1s
2026-01-03T13:09:35.850848Z  INFO runtime::init::task_history: Task history enabled: retention_period=28800s, retention_check_interval=900s
2026-01-03T13:09:35.851436Z  INFO runtime::flight: Spice Runtime Flight listening on 127.0.0.1:50051
2026-01-03T13:09:35.851441Z  INFO runtime::http: Spice Runtime HTTP listening on 127.0.0.1:8090
2026-01-03T13:09:35.852728Z  INFO runtime::init::dataset: Dataset part initializing...
2026-01-03T13:09:35.852727Z  INFO runtime::init::dataset: Dataset partsupp initializing...
2026-01-03T13:09:35.852739Z  INFO runtime::init::dataset: Dataset region initializing...
2026-01-03T13:09:35.852750Z  INFO runtime::init::dataset: Dataset customer initializing...
2026-01-03T13:09:35.852750Z  INFO runtime::init::dataset: Dataset supplier initializing...
2026-01-03T13:09:35.852754Z  INFO runtime::init::dataset: Dataset orders initializing...
2026-01-03T13:09:35.852755Z  INFO runtime::init::dataset: Dataset lineitem initializing...
2026-01-03T13:09:35.852755Z  INFO runtime::init::dataset: Dataset nation initializing...
2026-01-03T13:09:35.897909Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.898142Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.898943Z  INFO runtime::init::dataset: Dataset region registered (s3://benchmarks/tpch_sf1/region.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.898943Z  INFO runtime::init::dataset: Dataset part registered (s3://benchmarks/tpch_sf1/part.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.902178Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset part
2026-01-03T13:09:35.902490Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset region
2026-01-03T13:09:35.902925Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.903611Z  INFO runtime::init::dataset: Dataset partsupp registered (s3://benchmarks/tpch_sf1/partsupp.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.908175Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset partsupp
2026-01-03T13:09:35.915588Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.916460Z  INFO runtime::init::dataset: Dataset customer registered (s3://benchmarks/tpch_sf1/customer.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.919983Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset customer
2026-01-03T13:09:35.925062Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.926025Z  INFO runtime::init::dataset: Dataset supplier registered (s3://benchmarks/tpch_sf1/supplier.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.930007Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset supplier
2026-01-03T13:09:35.931415Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.932243Z  INFO runtime::init::dataset: Dataset orders registered (s3://benchmarks/tpch_sf1/orders.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.934675Z  INFO runtime::accelerated_table::refresh_task: Loaded 5 rows (1008.00 B) for dataset region in 32ms.
2026-01-03T13:09:35.936207Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset orders
2026-01-03T13:09:35.937686Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.938960Z  INFO runtime::init::dataset: Dataset lineitem registered (s3://benchmarks/tpch_sf1/lineitem.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.943595Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset lineitem
2026-01-03T13:09:35.944131Z  WARN runtime::init::dataset: Turso data accelerator (Alpha) is in preview and should not be used in production.
2026-01-03T13:09:35.945231Z  INFO runtime::init::dataset: Dataset nation registered (s3://benchmarks/tpch_sf1/nation.parquet), acceleration (turso:file), results cache enabled.
2026-01-03T13:09:35.948682Z  INFO runtime::accelerated_table::refresh_task: Loading data for dataset nation
2026-01-03T13:09:36.007356Z  INFO runtime::accelerated_table::refresh_task: Loaded 25 rows (3.35 kiB) for dataset nation in 58ms.
2026-01-03T13:09:36.145930Z  INFO runtime::accelerated_table::refresh_task: Loaded 10,000 rows (1.88 MiB) for dataset supplier in 215ms.
2026-01-03T13:09:37.009237Z  INFO runtime::accelerated_table::refresh_task: Loaded 150,000 rows (32.45 MiB) for dataset customer in 1s 89ms.
2026-01-03T13:09:37.392366Z  INFO runtime::accelerated_table::refresh_task: Loaded 200,000 rows (36.27 MiB) for dataset part in 1s 490ms.
2026-01-03T13:09:40.618599Z  INFO runtime::accelerated_table::refresh_task: Loaded 800,000 rows (134.02 MiB) for dataset partsupp in 4s 710ms.
2026-01-03T13:09:45.117395Z  INFO runtime::accelerated_table::refresh_task: Loaded 1,500,000 rows (222.91 MiB) for dataset orders in 9s 181ms.
2026-01-03T13:09:45.996978Z  INFO runtime::accelerated_table::refresh_task: Dataset lineitem received 1,380,927 records (272.00 MiB) in 10s, 27.17 MiB/s
2026-01-03T13:09:56.018948Z  INFO runtime::accelerated_table::refresh_task: Dataset lineitem received 2,888,255 records (568.13 MiB) in 20s, 28.36 MiB/s
2026-01-03T13:10:06.050741Z  INFO runtime::accelerated_table::refresh_task: Dataset lineitem received 4,403,775 records (865.94 MiB) in 30s, 28.80 MiB/s
2026-01-03T13:10:16.072268Z  INFO runtime::accelerated_table::refresh_task: Dataset lineitem received 5,952,063 records (1.14 GiB) in 40s, 29.19 MiB/s
2026-01-03T13:10:16.433116Z  INFO runtime::accelerated_table::refresh_task: Loaded 6,001,215 rows (1.15 GiB) for dataset lineitem in 40s 489ms.
2026-01-03T13:10:16.482547Z  INFO runtime: All components are loaded. Spice runtime is ready!
Running benchmark test






Error: Benchmark test failed due to failed queries:
tpch_q1: Explain plan snapshot assertion failed
tpch_q10: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: orders`", details: [], metadata: MetadataMap { headers: {} }
tpch_q11: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: supplier`", details: [], metadata: MetadataMap { headers: {} }
tpch_q12: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: orders`", details: [], metadata: MetadataMap { headers: {} }
tpch_q13: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: orders`", details: [], metadata: MetadataMap { headers: {} }
tpch_q14: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: part`", details: [], metadata: MetadataMap { headers: {} }
tpch_q16: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: part`", details: [], metadata: MetadataMap { headers: {} }
tpch_q17: Query execution failed: status: InvalidArgument, message: "Correlated scalar subquery can only be used in Projection, Filter, Aggregate plan nodes", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Sat, 03 Jan 2026 13:10:36 GMT"} }
tpch_q18: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: orders`", details: [], metadata: MetadataMap { headers: {} }
tpch_q19: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: part`", details: [], metadata: MetadataMap { headers: {} }
tpch_q2: Query execution failed: status: InvalidArgument, message: "Correlated scalar subquery can only be used in Projection, Filter, Aggregate plan nodes", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Sat, 03 Jan 2026 13:10:32 GMT"} }
tpch_q20: Query execution failed: status: InvalidArgument, message: "Correlated scalar subquery can only be used in Projection, Filter, Aggregate plan nodes", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Sat, 03 Jan 2026 13:10:36 GMT"} }
tpch_q21: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: lineitem`", details: [], metadata: MetadataMap { headers: {} }
tpch_q22: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: orders`", details: [], metadata: MetadataMap { headers: {} }
tpch_q3: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: orders`", details: [], metadata: MetadataMap { headers: {} }
tpch_q4: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `unexpected token ''3 MONS'' at offset 193`", details: [], metadata: MetadataMap { headers: {} }
tpch_q5: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: orders`", details: [], metadata: MetadataMap { headers: {} }
tpch_q6: Query validation failed: DataMismatch { column: "revenue", row_number: 1, expected: "\"123141078.2283\"", actual: "None" }
tpch_q7: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: lineitem`", details: [], metadata: MetadataMap { headers: {} }
tpch_q8: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: supplier`", details: [], metadata: MetadataMap { headers: {} }
tpch_q9: Failed to process query stream: Tonic error: status: InvalidArgument, message: "Turso query failed: SQL execution failure: `Parse error: no such table: supplier`", details: [], metadata: MetadataMap { headers: {} }
Error: Process completed with exit code 1.
```

--- a/crates/test-framework/src/queries/mod.rs
+++ b/crates/test-framework/src/queries/mod.rs
@@ -902,9 +902,14 @@ pub fn get_tpch_test_queries(overrides: Option<QueryOverrides>) -> Vec<Query> {
             let mut queries: Vec<Query> = remove_tpch_query!(
                 queries,
                 2, // Correlated scalar subquery not supported; DF limitation, Turso tests are not cross-table federated
+                4, // Federation fails for cross-provider subquery filters; https://github.com/spiceai/spiceai/issues/9879
                 6, // Rewritten: explicit BETWEEN 0.05 AND 0.07 to avoid libSQL float arithmetic precision issue; https://github.com/spiceai/spiceai/issues/9872
+                16, // Federation fails for cross-provider subquery filters; https://github.com/spiceai/spiceai/issues/9879
                 17, // Correlated scalar subquery not supported
-                20  // Correlated scalar subquery not supported
+                18, // Federation fails for cross-provider subquery filters; https://github.com/spiceai/spiceai/issues/9879
+                20, // Correlated scalar subquery not supported
+                21, // Federation fails for cross-provider subquery filters; https://github.com/spiceai/spiceai/issues/9879
+                22 // Federation fails for cross-provider subquery filters; https://github.com/spiceai/spiceai/issues/9879
             );
             queries.extend(generate_tpch_queries_override!("turso", q6));
             queries

--- a/tools/testoperator/dispatch/tpch/sf1/accelerated/file[parquet]-turso[file].yaml
+++ b/tools/testoperator/dispatch/tpch/sf1/accelerated/file[parquet]-turso[file].yaml
@@ -1,12 +1,11 @@
 tests:
-  # https://github.com/spiceai/spiceai/issues/8764
-  # bench:
-  #   spicepod_path: accelerated/file[parquet]-turso[file].yaml
-  #   query_set: tpch
-  #   query_overrides: turso
-  #   runner_type: spiceai-dev-runners
-  #   ready_wait: 600
-  #   validate_results: true
+  bench:
+    spicepod_path: accelerated/file[parquet]-turso[file].yaml
+    query_set: tpch
+    query_overrides: turso
+    runner_type: spiceai-dev-runners
+    ready_wait: 600
+    validate_results: true
   throughput:
     spicepod_path: accelerated/file[parquet]-turso[file].yaml
     query_overrides: turso

